cmd/nerdctl/main.go

```diff
@@ -440,36 +421,3 @@ func AddPersistentBoolFlag(cmd *cobra.Command, name string, aliases, nonPersiste
 		}
 	}
 }
-
-// AddPersistentStringArrayFlag is similar to cmd.Flags().StringArray but supports aliases and env var and persistent.
```

Comment:
AddPersistentStringFlag should be moved too

Reply:
We should, of course.
It is only used by `main.go` though (and so is `AddPersistentBoolFlag`).
There is also a fair amount of cleanup to do in `helpers`.

What about we take care of these (and further scrubbing the new helpers directory) in a follow-up PR?

Reply:
Changed my mind with more coffee beating the lazyness out of me.

Added a new commit that take care of post-cleanup:
- moved the last Flag methods
- moved some linux specific stuff from `testing.go` to `testing_linux.go`
- removed now empty `main_unix.go`